### PR TITLE
remove unused/wrong/broken plugins

### PR DESCRIPTION
**Pull request by psychocoderHPC**
I am currently working on plugin support for multi-species, therefor I must touch most of the plugin. To minimize the upcoming pull request I will remove some plugins.

@ComputationalRadiationPhysics/picongpu-developers 
Please look to the list of removed plugins, if you think I removed to much please write here and I can delete the commit for the plugin. If you are fine with the list than also write it down.

**Removed plugins:**
- FieldEnergy
  - only 3D support
  - used only sequential float_X reduce
  - one file per step
  - issue #324
- ~~IntensityPlugin~~
  - ~~only 3D support~~
  - ~~open problems: #327~~
- ParticleSpectrum
  - only 3D support
  - not implemented `notify` method #769 
- ~~LiveViewPlugin~~
  - ~~server not released~~
  - ~~IMO not really needed~~
- ~~SumCurrents~~
  - ~~only output per process on std::cout~~
- BinaryDensity
  - only float output (do to gnuplot restriction)
  - output special for gnuplot
  - only 3D support
- ParticleDensity
  - only 3D support
  - only in counter per cell
  - fast int counter overflow

### CODE REVIEW

**Comment by PrometheusPi** [comment]
- I agree, we should **remove** `FieldEnergy` since the bugs in `EnergyField` are fixed with #502.
 - The output of `IntensityPlugin` would be usefull to analyze plama instabilities, since inplementing this by reading `hdf5` dumps is extremly slow. But it does not work in its current state. I am pro **fixing** this plugin.
 - I agree that `ParticleSpectrum` should be **removed** since it is a duplication of `BinEnergyParticles` but does not work.
 - I agree that `LiveViewPlugin`should be **removed** since we have `insituvolvis`.
 - `SumCurrents` is probably not usefull in its current state - therefore **remove**.
 - I am not familiar with `BinaryDensity` - **no opinion**. 
 - `ParticleDensity` is covered by `hdf5` - **remove**.

**Comment by ax3l** [comment]
I agree with @PrometheusPi 
- `IntensityPlugin`: fix

I would also vote to keep
- `LiveViewPlugin` - it's not the same as the 3D in-situ vis and connecting a client should be possible and simply needs to be done

- `SumCurrents`: fix, just add an MPI_Reduce there. It's very useful for single particle tests - keep it.

Ok to remove the rest, pls also remove the scripts for `ParticleDensity`/`BinaryDensity` in `src/tools`.

**Comment by PrometheusPi** [comment]
I agree with @ax3l: **renaming** of `IntensityPlugin` is necessary as well.

**Comment by psychocoderHPC** [comment]
**IntensityPlugin:**
I agree that we need a plugin which does nearly the same like `IntensityPlugin` but if we rename the plugin and afterwards rewrite nearly all we can remove this plugin and wrote a new one. This is easier to review than a rewrite of min. 80% of the plugin. My suggestion open an issue and remove the old plugin.

**LiveViewPlugin:** 
We can bring back this plugin if someone had time to write a server but up to this moment this plugin is useless. 

**SumCurrents**
We can keep this, but we need to rename it. It is not the "sum of current" it is the sum over the current change.

**Comment by ax3l** [comment]
:+1:

**Comment by psychocoderHPC** [comment]
I removed the plugin `SumCurrent` from this pull request.

**Comment by ax3l** [comment]
You are still removing `IntensityPlugin` in this pull. Pls revert that change.

**Comment by psychocoderHPC** [comment]
removed plugin from this pull request:
- `LiveViewPlugin`
- `IntensityPlugin`

**Comment by ax3l** [comment]
Current change in plugins:

- FieldEnergy
- ParticleDensity
- BinaryDensity
- ParticleSpectrum

looks ok - I'll just wait for the CI to confirm.

File: src/picongpu/include/plugins/PluginController.hpp [hunk 1]
```diff
@@ -114,25 +105,19 @@ class PluginController : public ILightweightPlugin
...
 #endif
-    typedef ParticleDensity<PIC_Electrons, DensityToBinary, float_X> ElectronsBinaryDensityBuilder;
 
 #if(ENABLE_HDF5 == 1)
     /* speciesParticleShape::ParticleShape::ChargeAssignment */
     typedef PhaseSpaceMulti<particles::shapes::Counter::ChargeAssignment, PIC_Electrons> PhaseSpaceElectrons;
 #endif
 #if(SIMDIM==DIM3)
-#if(PIC_ENABLE_PNG==1)
-    typedef heiko::ParticleDensity<PIC_Electrons> HeikoParticleDensity;
```
Comment (ax3l) [inline src/picongpu/include/plugins/PluginController.hpp]: @f-schmitt-zih we just removed the `heiko::` namespace. can you believe that? ;)

**Pull request merged**